### PR TITLE
Fix erroneous "did not start playing recording" messages

### PR DESCRIPTION
**Pull request by ctrlaltdavid**

### CODE REVIEW

**Comment by ctrlaltdavid** [comment]
Testing

Load and unload a recording a few times. You should not get erroneous "Didn't start playing recording" error dialogs.

**Comment by hifi-gustavo** [comment]
Build console output for commit: c2d22a1246248e49b9454449df84a6483e02bbf3 available at: https://gist.github.com/d66cc94c9e5e49193af88e22207e22d3<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10427-c2d22a1.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10427-c2d22a1.dmg

**Comment by anshuman64** [comment]
QA Fail.

Repro:
1. Have 5 AC's
2. Create a recording
3. Load 3 of that recording (total of 4)
4. Unload all 4 recordings
Result: Most likely will get erroneous "Didn't start playing recording ___" message
Then try to load another recording. May or may not begin playing.

File: scripts/system/playRecordingAC.js [hunk 1]
```diff
@@ -368,13 +368,16 @@
...
             playing: Player.isPlaying(),
             recording: Player.recording(),
             entity: Entity.id()
         }));
-        heartbeatTimer = Script.setTimeout(sendHeartbeat, HEARTBEAT_INTERVAL);
+
+        if (!isOneShot) {
+            heartbeatTimer = Script.setTimeout(sendHeartbeat, HEARTBEAT_INTERVAL);
+        }
     }
```
Comment (AndrewMeadows) [inline scripts/system/playRecordingAC.js]: Not a blocker, but an idea on how to make the code more readable:

(1) Modify `sendHeartbeat()` to do only that: send one heartbeat.
(2) Add a method called `startHeartbeat()` which sets the timeout interval.
(3) Call the right methods in the right places.

**Comment by anshuman64** [comment]
QA Fail. Repro:

1. Have 4 AC's running
2. Create/Load 4 recordings
3. Unload all 4 recordings. Sometimes, I erroneously get the error message here. But the recordings unload properly.
4. Load 4 recordings. Very frequently, I erroneously get the error message here. But the recordings load properly (which is an improvement).
5. Unload all 4 recordings. Sometimes, I erroneously get the error message here. Also, if I wait a while, some of the recordings I unloaded start up again.

I understand this is a complicated issue to resolve given our roundabout implementation. I think these changes are an improvement, because at least the recordings load when they are supposed to. Even though they are not unloading when they are supposed to, that's not too annoying as you can just press the "X" again and it usually works the second time. 

Bottom line, let me know if fixing this is getting too annoying and we can pick the least evil of all bugs and move on for now.

**Comment by ctrlaltdavid** [comment]
Testing
"Cannot play / load recording" messages should only occur if AC script cannot (a) create an entity to persist the recording to, or (b) fails to load the recording from ATP.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2bed69cfea194ec1315861b458ad7f9d20518730 available at: https://gist.github.com/d66cc94c9e5e49193af88e22207e22d3<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10427-2bed69c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10427-2bed69c.dmg

**Comment by anshuman64** [comment]
QA Fail. Repro:
1. Have 4 AC's running.
2. Create 1 recording.
3. Load that recording 3 more times.
4. Unload all recordings.
5. Try loading that recording again multiple times.
Result: Sometimes the recording loads fine. Sometimes the recording does not load, gives an error dialog, and still shows up in the table. Sometimes the recording does not load, gives NO error dialog, and still shows up in the table.

However, behavior has improved since before. We can merge this PR and create separate tickets for the problems I continue to see.

**Comment by anshuman64** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0cd5fe35da64b1f83442864e580cac48c8069f71 available at: https://gist.github.com/d66cc94c9e5e49193af88e22207e22d3<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10427-0cd5fe3.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10427-0cd5fe3.dmg

**Comment by anshuman64** [comment]
This is much more stable! Thanks David for looking into it. Giving this a well deserved QA Pass :)
